File: tesodev_api_customers/models/Address.js
const mongoose = require("mongoose");

const AddressSchema = new mongoose.Schema({
  addressLine: String,
  city: {
    type: String,
    required: true,
  },
  country: {
    type: String,
    required: true,
  },
  cityCode: {
    type: Number,
    required: true,
  },
});

module.exports = AddressSchema;

File: demo/src/main/java/com/example/demo/query/entity/Address.java
package com.example.demo.query.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.Valid;
import javax.validation.constraints.NotBlank;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Valid
public class Address {

    private String addressLine = "";
    @NotBlank
    private String city = "";
    @NotBlank
    private String country = "";
    @NotBlank
    private String cityCode = "";
}
